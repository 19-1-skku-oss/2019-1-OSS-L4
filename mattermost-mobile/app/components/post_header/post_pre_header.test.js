// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import PostPreHeader from './post_pre_header';

describe('PostPreHeader', () => {
    const baseProps = {
        isConsecutive: false,
        isFlagged: false,
        isPinned: false,
        rightColumnStyle: [],
        skipFlaggedHeader: false,
        skipPinnedHeader: false,
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot when not flagged or pinned post', () => {
        const wrapper = shallow(
            <PostPreHeader {...baseProps}/>
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.type()).toBeNull();
    });

    test('should match snapshot when flagged post is set but skipFlaggedHeader is true', () => {
        const props = {
            ...baseProps,
            isFlagged: true,
            skipFlaggedHeader: true,
        };

        const wrapper = shallow(
            <PostPreHeader {...props}/>
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.type()).toBeNull();
    });

    test('should match snapshot when pinned post is set but skipPinnedHeader is true', () => {
        const props = {
            ...baseProps,
            isPinned: true,
            skipPinnedHeader: true,
        };

        const wrapper = shallow(
            <PostPreHeader {...props}/>
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.type()).toBeNull();
    });

    test('should match snapshot when flagged and not pinned', () => {
        const props = {
            ...baseProps,
            isFlagged: true,
        };

        const wrapper = shallow(
            <PostPreHeader {...props}/>
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find('#flagIcon').exists()).toEqual(true);
        expect(wrapper.find('#pinIcon').exists()).toEqual(false);
        expect(wrapper.find('FormattedText').first().props().id).toEqual('mobile.post_pre_header.flagged');
    });

    test('should match snapshot when pinned and not flagged', () => {
        const props = {
            ...baseProps,
            isPinned: true,
        };

        const wrapper = shallow(
            <PostPreHeader {...props}/>
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find('#flagIcon').exists()).toEqual(false);
        expect(wrapper.find('#pinIcon').exists()).toEqual(true);
        expect(wrapper.find('FormattedText').first().props().id).toEqual('mobile.post_pre_header.pinned');
    });

    test('should match snapshot when pinned and flagged', () => {
        const props = {
            ...baseProps,
            isFlagged: true,
            isPinned: true,
        };

        const wrapper = shallow(
            <PostPreHeader {...props}/>
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find('#flagIcon').exists()).toEqual(true);
        expect(wrapper.find('#pinIcon').exists()).toEqual(true);
        expect(wrapper.find('FormattedText').first().props().id).toEqual('mobile.post_pre_header.pinned_flagged');
    });

    test('should match snapshot when pinned and flagged but skipping pinned', () => {
        const props = {
            ...baseProps,
            isFlagged: true,
            isPinned: true,
            skipPinnedHeader: true,
        };

        const wrapper = shallow(
            <PostPreHeader {...props}/>
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find('#flagIcon').exists()).toEqual(true);
        expect(wrapper.find('#pinIcon').exists()).toEqual(false);
        expect(wrapper.find('FormattedText').first().props().id).toEqual('mobile.post_pre_header.flagged');
    });

    test('should match snapshot when pinned and flagged but skipping flagged', () => {
        const props = {
            ...baseProps,
            isFlagged: true,
            isPinned: true,
            skipFlaggedHeader: true,
        };

        const wrapper = shallow(
            <PostPreHeader {...props}/>
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find('#flagIcon').exists()).toEqual(false);
        expect(wrapper.find('#pinIcon').exists()).toEqual(true);
        expect(wrapper.find('FormattedText').first().props().id).toEqual('mobile.post_pre_header.pinned');
    });

    test('should match snapshot when pinned and flagged but skipping both', () => {
        const props = {
            ...baseProps,
            isFlagged: true,
            isPinned: true,
            skipFlaggedHeader: true,
            skipPinnedHeader: true,
        };

        const wrapper = shallow(
            <PostPreHeader {...props}/>
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.type()).toBeNull();
    });
});
