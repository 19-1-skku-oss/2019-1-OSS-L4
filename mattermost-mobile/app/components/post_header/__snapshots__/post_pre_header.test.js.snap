// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostPreHeader should match snapshot when flagged and not pinned 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
        "height": 15,
        "marginLeft": 12,
        "marginRight": 10,
        "marginTop": 10,
      },
      false,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginRight": 10,
        "width": 35,
      }
    }
  >
    <Image
      id="flagIcon"
      source={
        Object {
          "testUri": "../../../dist/assets/images/post_header/flag.png",
        }
      }
      style={
        Object {
          "height": 11,
          "tintColor": "#2389d7",
          "width": 11,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Array [],
      ]
    }
  >
    <FormattedText
      defaultMessage="Flagged"
      id="mobile.post_pre_header.flagged"
      style={
        Object {
          "color": "#2389d7",
          "fontSize": 13,
          "lineHeight": 15,
        }
      }
    />
  </View>
</View>
`;

exports[`PostPreHeader should match snapshot when flagged post is set but skipFlaggedHeader is true 1`] = `null`;

exports[`PostPreHeader should match snapshot when not flagged or pinned post 1`] = `null`;

exports[`PostPreHeader should match snapshot when pinned and flagged 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
        "height": 15,
        "marginLeft": 12,
        "marginRight": 10,
        "marginTop": 10,
      },
      false,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginRight": 10,
        "width": 35,
      }
    }
  >
    <Image
      id="pinIcon"
      source={
        Object {
          "testUri": "../../../dist/assets/images/post_header/pin.png",
        }
      }
      style={
        Object {
          "height": 11,
          "tintColor": "#2389d7",
          "width": 11,
        }
      }
    />
    <View
      style={
        Object {
          "marginRight": 5,
        }
      }
    />
    <Image
      id="flagIcon"
      source={
        Object {
          "testUri": "../../../dist/assets/images/post_header/flag.png",
        }
      }
      style={
        Object {
          "height": 11,
          "tintColor": "#2389d7",
          "width": 11,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Array [],
      ]
    }
  >
    <FormattedText
      defaultMessage="Pinned and Flagged"
      id="mobile.post_pre_header.pinned_flagged"
      style={
        Object {
          "color": "#2389d7",
          "fontSize": 13,
          "lineHeight": 15,
        }
      }
    />
  </View>
</View>
`;

exports[`PostPreHeader should match snapshot when pinned and flagged but skipping both 1`] = `null`;

exports[`PostPreHeader should match snapshot when pinned and flagged but skipping flagged 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
        "height": 15,
        "marginLeft": 12,
        "marginRight": 10,
        "marginTop": 10,
      },
      false,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginRight": 10,
        "width": 35,
      }
    }
  >
    <Image
      id="pinIcon"
      source={
        Object {
          "testUri": "../../../dist/assets/images/post_header/pin.png",
        }
      }
      style={
        Object {
          "height": 11,
          "tintColor": "#2389d7",
          "width": 11,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Array [],
      ]
    }
  >
    <FormattedText
      defaultMessage="Pinned"
      id="mobile.post_pre_header.pinned"
      style={
        Object {
          "color": "#2389d7",
          "fontSize": 13,
          "lineHeight": 15,
        }
      }
    />
  </View>
</View>
`;

exports[`PostPreHeader should match snapshot when pinned and flagged but skipping pinned 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
        "height": 15,
        "marginLeft": 12,
        "marginRight": 10,
        "marginTop": 10,
      },
      false,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginRight": 10,
        "width": 35,
      }
    }
  >
    <Image
      id="flagIcon"
      source={
        Object {
          "testUri": "../../../dist/assets/images/post_header/flag.png",
        }
      }
      style={
        Object {
          "height": 11,
          "tintColor": "#2389d7",
          "width": 11,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Array [],
      ]
    }
  >
    <FormattedText
      defaultMessage="Flagged"
      id="mobile.post_pre_header.flagged"
      style={
        Object {
          "color": "#2389d7",
          "fontSize": 13,
          "lineHeight": 15,
        }
      }
    />
  </View>
</View>
`;

exports[`PostPreHeader should match snapshot when pinned and not flagged 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
        "height": 15,
        "marginLeft": 12,
        "marginRight": 10,
        "marginTop": 10,
      },
      false,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginRight": 10,
        "width": 35,
      }
    }
  >
    <Image
      id="pinIcon"
      source={
        Object {
          "testUri": "../../../dist/assets/images/post_header/pin.png",
        }
      }
      style={
        Object {
          "height": 11,
          "tintColor": "#2389d7",
          "width": 11,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Array [],
      ]
    }
  >
    <FormattedText
      defaultMessage="Pinned"
      id="mobile.post_pre_header.pinned"
      style={
        Object {
          "color": "#2389d7",
          "fontSize": 13,
          "lineHeight": 15,
        }
      }
    />
  </View>
</View>
`;

exports[`PostPreHeader should match snapshot when pinned post is set but skipPinnedHeader is true 1`] = `null`;
